python/whylogs/core/column_profile.py

```diff
@@ -49,14 +49,21 @@ def flush(self) -> None:
         self._cache = []
         self.track_column(old_cache)
 
-    def track_column(self, series: Any) -> None:
-        ex_col = PreprocessedColumn.apply(series)
+    def _process_extracted_column(self, extracted_column: PreprocessedColumn) -> None:
         for metric in self._metrics.values():
-            res = metric.columnar_update(ex_col)
+            res = metric.columnar_update(extracted_column)
             self._failure_count += res.failures
             self._success_count += res.successes
         for validator in self._column_validators:
-            validator.columnar_validate(ex_col)
+            validator.columnar_validate(extracted_column)
+
+    def track_column(self, series: Any) -> None:
+        ex_col = PreprocessedColumn.apply(series)
+        self._process_extracted_column(ex_col)
+
+    def _track_datum(self, value: Any) -> None:
```

Comment:
why is this private if it's used outside of this module?

Reply:
its internal to whylogs, so telegraphing that community users should not consider this part of the public API, and we will likely change this a bit as we optimize and do more testing, but wanted to get this out first in the shape we tested it.